core/src/main/scala/com/salesforce/op/filters/RawFeatureFilter.scala

```diff
@@ -192,23 +192,24 @@ class RawFeatureFilter[T]
 
   /**
    * Take in the distribution summaries for datasets (scoring summary may be empty) and determine which
-   * features should be dropped (including maps with all keys dropped) and which map keys need to be dropped
+   * features should be dropped (including maps with all keys dropped), which map keys need to be dropped, and
+   * why those features are dropped
    *
    * @param trainingDistribs summary of distributions for training data features
    * @param scoringDistribs  summary of distributions for scoring data features (may be an empty seq)
    * @param correlationInfo  info needed to determine feature to drop based on null label-leakage correlation
-   * @return a list of feature names that should be dropped and a map of map keys that should be dropped
-   *         Map(featureName -> key)
+   * @return a list of outcomes of raw feature filter tests for each feature, a list of feature names that
+   *         should be dropped and a map of map keys that should be dropped Map(featureName -> key)
    */
   private[op] def getFeaturesToExclude(
     trainingDistribs: Seq[FeatureDistribution],
     scoringDistribs: Seq[FeatureDistribution],
     correlationInfo: Map[FeatureKey, Map[FeatureKey, Double]]
-  ): (Seq[String], Map[String, Set[String]]) = {
+  ): (Seq[ExclusionReasons], Seq[String], Map[String, Set[String]]) = {
```

Comment:
I think we are at point where we should introduce a case class for this `case class RawFeatureFilterOutcomes(exclusionReasons, ... etc)`